src/blockchainApiConnection/__mocks__/BlockchainApiConnection.ts

```diff
@@ -5,12 +5,141 @@
+    )
+  }
+  return BlockchainApiConnection.instance
 }
 
-export default getCached
+const TxresultsQueue: SubmittableResult[] = []
+let defaultTxResult: SubmittableResult = __makeSubmittableResult(true)
+
+class MockSubmittableExtrinsic {
+  result: SubmittableResult
+  method: string = 'mock tx'
+
+  constructor(result: SubmittableResult) {
+    this.result = result
+  }
+
+  public sign(_a: any, _b: any, _c: any) {
+    return this
+  }
+
+  public send(callable: Function) {
+    if (callable) {
+      callable(this.result)
+    }
+  }
+}
+
+function __getMockSubmittableExtrinsic(): SubmittableExtrinsic {
+  const result: SubmittableResult = TxresultsQueue.shift() || defaultTxResult
+  return (new MockSubmittableExtrinsic(result) as any) as SubmittableExtrinsic
+}
+
+function __makeSubmittableResult(success: boolean): SubmittableResult {
+  const status: ExtrinsicStatus = {
+    type: success ? 'Finalized' : 'Invalid',
+    isFinalized: success,
+    isDropped: false,
+    isInvalid: !success,
+    isUsurped: false,
+    isFuture: false,
+    isReady: true,
+  } as any
+
+  return new SubmittableResult({
+    status,
+  })
+}
+
+function __queueResults(results: boolean[]) {
+  results.forEach(success => {
+    TxresultsQueue.push(__makeSubmittableResult(success))
+  })
+}
+
+function __setDefaultResult(success: boolean) {
+  defaultTxResult = __makeSubmittableResult(success)
+}
+
+const __mocked_api: any = {
+  tx: {
+    attestation: {
+      add: jest.fn((claimHash, _cTypeHash) => {
+        return __getMockSubmittableExtrinsic()
+      }),
+    },
+    balances: {
+      transfer: jest.fn(() => __getMockSubmittableExtrinsic()),
+    },
+    ctype: {
+      add: jest.fn((hash, signature) => {
+        return __getMockSubmittableExtrinsic()
+      }),
+    },
+    delegation: {
+      createRoot: jest.fn((rootId, _ctypeHash) => {
+        return __getMockSubmittableExtrinsic()
+      }),
+      revokeRoot: jest.fn(rootId => {
+        return __getMockSubmittableExtrinsic()
+      }),
+      revokeDelegation: jest.fn(delegationId => {
+        return __getMockSubmittableExtrinsic()
+      }),
+    },
+    did: {
+      add: jest.fn((sign_key, box_key, doc_ref) => {
+        return __getMockSubmittableExtrinsic()
+      }),
+      remove: jest.fn(() => {
+        return __getMockSubmittableExtrinsic()
+      }),
+    },
+  },
+  query: {
+    system: { accountNonce: jest.fn(() => new Text('12345')) },
+    attestation: {
+      delegatedAttestations: jest.fn(),
+      attestations: jest.fn(),
+    },
+    balances: {
+      freeBalance: jest.fn((account: string) => new BN(0)),
+    },
+    ctype: {
+      cTYPEs: jest.fn(hash => true),
+    },
+    delegation: {
+      root: jest.fn((rootId: string) => new Option(Tuple)),
+      delegations: jest.fn((delegationId: string) => new Option(Tuple)),
+      children: jest.fn((id: string) => new Vec(Text, [])),
+    },
+    did: {
+      dIDs: jest.fn(id => new Option(Tuple)),
```

Comment:
I will submit a proposal for a default query here. There, one can add explanatory comments and reference to this in custom mocks.

Reply:
My approach was to return null result queries by default (what would be returned by a query for something that is not there), because it seems odd to return a specific result as a default value. Would prefer to stick to that approach, but you're right that documentation and example cases must be given. I could comment this out and leave it as an example return value, and also add similar examples to the other queries?

Reply:
Makes sense and I like your proposal.

Reply:
As for documentation on how to set return values, I could also add a comment at the top of the mock or file that illustrates this once.

Reply:
Sounds good if it's not too obvious as in this case!